core/src/main/java/com/linecorp/armeria/common/stream/DefaultStreamMessageDuplicator.java

```diff
@@ -244,6 +243,7 @@ private void doPushSignal(Object obj) {
                     final int allowedMaxSignalLength = maxSignalLength - signalLength;
                     if (dataLength > allowedMaxSignalLength) {
                         final ContentTooLargeException cause = ContentTooLargeException.get();
+                        StreamMessageUtil.closeOrAbort(obj, cause);
```

Comment:
Similarly, wouldn't `PooledObjects.close()` be enough?

Reply:
I know most users only use this `DefaultStreamMessageDuplicator` via `Http{Request,Response}.duplicate()`.
However, the type parameter `T` of `DefaultStreamMessageDuplicator` is unbounded, and `DefaultStreamMessageDuplicator` has a public constructor.
I thought the class could handle other types by users just in case.